pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -146,53 +152,88 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 			Type:         resourceName,
 		})
 
+		podReusableMemory := p.getPodReusableMemory(pod, bestHint.NUMANodeAffinity, resourceName)
+		if podReusableMemory >= requestedSize {
+			requestedSize = 0
+		} else {
+			requestedSize -= podReusableMemory
+		}
+
 		// Update nodes memory state
-		for _, nodeID := range maskBits {
-			machineState[nodeID].NumberOfAssignments++
-			machineState[nodeID].Cells = maskBits
+		p.updateMachineState(machineState, maskBits, resourceName, requestedSize)
+	}
 
-			// we need to continue to update all affinity mask nodes
-			if requestedSize == 0 {
-				continue
-			}
+	p.updatePodReusableMemory(pod, container, containerBlocks)
 
-			// update the node memory state
-			nodeResourceMemoryState := machineState[nodeID].MemoryMap[resourceName]
-			if nodeResourceMemoryState.Free <= 0 {
-				continue
-			}
+	s.SetMachineState(machineState)
+	s.SetMemoryBlocks(podUID, container.Name, containerBlocks)
 
-			// the node has enough memory to satisfy the request
-			if nodeResourceMemoryState.Free >= requestedSize {
-				nodeResourceMemoryState.Reserved += requestedSize
-				nodeResourceMemoryState.Free -= requestedSize
-				requestedSize = 0
-				continue
-			}
+	// update init containers memory blocks to reflect the fact that we re-used init containers memory
+	// it possible that the size of the init container memory block will have 0 value, when all memory
+	// allocated for it was re-used
+	p.updateInitContainersMemoryBlocks(s, pod)
```

Comment:
Can you expand this comment a bit. Saying that we only do this so that the sum(memory_for_all_containers) == total amount of allocated memory to the pod (even though the fine state here doesn't accurately reflect what was (in reality) allocated to each container). Then add a TODO to update the data structure to handle this better in the future.

Reply:
done